diff --git a/build/config/android/BUILD.gn b/build/config/android/BUILD.gn
--- a/build/config/android/BUILD.gn
+++ b/build/config/android/BUILD.gn
@@ -55,6 +55,10 @@ config("compiler") {
     }
   }
 
+  # 16KB page size support for Android 15+ devices
+  # https://developer.android.com/guide/practices/page-sizes
+  ldflags += [ "-Wl,-z,max-page-size=16384" ]
+
   # Instead of using an unwind lib from the toolchain,
   # buildtools/third_party/libunwind will be built and used directly.
   ldflags += [ "--unwindlib=none" ]
